File: python/MissingNumber.py
def missing(a, n): 
    i, sum = 0, 1
 
    for i in range(2, n + 2):
        sum += i
        sum -= a[i - 2]
    return sum
 
arr = [1, 2, 3, 4, 6]
arr.sort()
print(missing(arr, len(arr)))

File: java/MissingNumber.java
public class Algorithms{
    static int missing(int a[], int n) {
        int sum = 1;
        for (int i = 2;i <= (n + 1);i++)
        {
            sum += i;
            sum -= a[i - 2];
        }
        return sum;
    }
    public static void main(String[] args)
    {
        int[] arr = { 1, 2, 4, 5 ,6};
        System.out.println(missing(arr, arr.length));
    }
}
 